main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,58 @@
+#include "main/lsp/requests/go_to_implementation.h"
+#include "core/lsp/QueryResponse.h"
+#include "main/lsp/json_types.h"
+#include "main/lsp/lsp.h"
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+
+GoToImplementationTask::GoToImplementationTask(const LSPConfiguration &config, MessageId id,
+                                               std::unique_ptr<ImplementationParams> params)
+    : LSPRequestTask(config, move(id), LSPMethod::TextDocumentImplementation), params(move(params)) {}
+
+unique_ptr<ResponseMessage> GoToImplementationTask::runRequest(LSPTypecheckerDelegate &typechecker) {
+    auto response = make_unique<ResponseMessage>("2.0", id, LSPMethod::TextDocumentImplementation);
+
+    const core::GlobalState &gs = typechecker.state();
+    auto queryResult =
+        queryByLoc(typechecker, params->textDocument->uri, *params->position, LSPMethod::TextDocumentImplementation);
+
+    if (queryResult.error) {
+        // An error happened while setting up the query.
+        response->error = move(queryResult.error);
+        return response;
+    }
+
+    vector<unique_ptr<Location>> result;
+    for (auto const &queryResponse : queryResult.responses) {
+        auto def = queryResponse->isDefinition();
```

Comment:
I think you should also be able to support go to implementation from:
* Abstract class definition sites
* Abstract class/module references
* Abstract method call sites.

As-written, this implementation only supports go to implementation from abstract method definition sites.

Reply:
Added the support for this use cases in the last commit. Please take a look!